functors_task_1/student/src/impl.cpp

```diff
@@ -1,2 +1,44 @@
 #include "impl.h"
 
+#include <algorithm>
+#include <forward_list>
+#include <map>
+#include <utility>
+
+BeerBrand BeerOrganizer::operator()()
+{
+    if (m_nextBeer <= static_cast<BaseType>(BeerBrand::None)
+            || (m_nextBeer >= static_cast<BaseType>(BeerBrand::Max)))
+    {
+        m_nextBeer = static_cast<BaseType>(BeerBrand::None) + 1;
+    }
+
+    return static_cast<BeerBrand>(m_nextBeer++);
+}
+
+bool isExpensiveBeer(const BeerBrand brand)
+{
+    //TODO: move it somewhere
+    static std::forward_list<BeerBrand> expensiveBeer = {BeerBrand::Corona, BeerBrand::HoeGaarden};
+    return std::find(std::begin(expensiveBeer), std::end(expensiveBeer), brand) != std::end(expensiveBeer);
+}
+
+Cocktail MixingPolicy::mix(const AlcoholDrink lhd, const NonAlcoholDrink rhd)
+{
+    //TODO: move it somewhere
+    static std::map<std::pair<AlcoholDrink, NonAlcoholDrink>, Cocktail> recipes = {
```

Comment:
1. Yep, mapping is a good pattern and should be used here, but please, implement this task without containers usage. You can easily replace it with a switch.
2. Why not const?

Reply:
p.s also, you don't need ordered container here :)

Reply:
Should be const and unordered.
The same about containers as in the comment above nothing said about it in the task description. :)